src/services/ui/src/api/submissionService.ts

```diff
@@ -97,32 +109,61 @@ export const submit = async <T extends Record<string, unknown>>({
   data,
   endpoint,
   user,
-  authority
+  authority,
 }: SubmissionServiceParameters<T>): Promise<SubmissionServiceResponse> => {
   if (data?.attachments) {
-    const attachments = data.attachments as AttachmentsSchema;
-    const validFilesetKeys = Object.keys(attachments).filter(key => attachments[key] !== undefined);
-    // TODO: Can this be a GET instead of POST w/ empty body
-    const preSignedURLs: PreSignedURL[] = await Promise.all(validFilesetKeys.map(() =>
-      API.post("os", "/getUploadUrl", {
-        body: {},
-      })
-    ));
-    const uploadRecipes: UploadRecipe[] = preSignedURLs.map((obj, idx) => ({
-      ...obj,
-      data: attachments[validFilesetKeys[idx]] as File[],
-      // Add your attachments object key and file label value to the attachmentTitleMap
-      // for this transform to work. Else the title will just be the object key.
-      title: attachmentTitleMap?.[validFilesetKeys[idx]] || validFilesetKeys[idx]
-    }));
+    // Drop nulls and non arrays
+    const validAttachmentSets = Object.fromEntries(
+      Object.entries(data.attachments).filter(([, value]) => value !== null)
+    );
+
+    // Generate a presigned url for each attachment
+    const preSignedURLs: PreSignedURL[] = await Promise.all(
+      Object.values(validAttachmentSets)
+        .flat()
+        .map(() =>
+          API.post("os", "/getUploadUrl", {
+            body: {},
+          })
+        )
+    );
+
+    // For each attachment, add name, title, and a presigned url... and push to uploadRecipes
+    const uploadRecipes: UploadRecipe[] = [];
+    let idx = 0;
+    for (const key in validAttachmentSets) {
+      if (Array.isArray(validAttachmentSets[key])) {
+        validAttachmentSets[key].forEach((obj: File) => {
+          uploadRecipes.push({
+            data: obj,
+            name: obj.name,
```

Comment:
This is messy, which prompted the idea above for just getting the attachment object key and file pairings handled when reducing the object down to its valid attachments above. If we did that, this would be much easier:

```typescript
const uploadRecipes: UploadRecipe[] = preSignedURLs.map((obj, idx) => ({
      ...obj,
      // Known to be undefined, from filtered array
      data: validAttachmentSets[idx]!.file,
      // Add your attachments object key and file label value to the attachmentTitleMap
      // for this transform to work. Else the title will just be the object key.
      title: attachmentTitleMap?.[validAttachmentSets[idx].attachmentKey] || validAttachmentSets[idx].attachmentKey,
      name: validAttachmentSets[idx].attachmentKey
    }));
```

I do think this should be changed before merge